src/Model/File.php

```diff
@@ -122,6 +167,125 @@ public function createFromPath(string $path, string $fileName = null): Model
 
         $entity->save();
 
+        // create thumbnail images if needed and possible
+        if ($entity->get('meta_is_image') && $this->createThumbnail === true) {
+            $entity->createThumbnail($path);
+        }
+
         return $entity;
     }
+
+    /**
+     * Create thumbnail images of current image.
+     *
+     * @param string $path Local path to original file
+     *
+     * @return bool True on success
+     */
+    public function createThumbnail(string $path): bool
+    {
+        $this->assertIsEntity();
+
+        $max_width = $this->thumbnailMaxWidth;
+        $max_height = $this->thumbnailMaxHeight;
+
+        [$width, $height, $image_type] = getimagesize($path);
+
+        switch ($image_type) {
+            case \IMAGETYPE_GIF:
+                $src = imagecreatefromgif($path);
+
+                break;
+            case \IMAGETYPE_JPEG:
+                $src = imagecreatefromjpeg($path);
+
+                break;
+            case \IMAGETYPE_PNG:
+                $src = imagecreatefrompng($path);
+
+                break;
+            default:
+                return false; // unsupported image type
+        }
+
+        $x_ratio = $max_width / $width;
+        $y_ratio = $max_height / $height;
+
+        if (($width <= $max_width) && ($height <= $max_height)) {
+            $tn_width = $width;
+            $tn_height = $height;
+        } elseif (($x_ratio * $height) < $max_height) {
+            $tn_height = (int) ceil($x_ratio * $height);
+            $tn_width = $max_width;
+        } else {
+            $tn_width = (int) ceil($y_ratio * $width);
+            $tn_height = $max_height;
+        }
+
+        $tmp = imagecreatetruecolor($tn_width, $tn_height);
+
+        // Check if this image is PNG or GIF, then set if Transparent
+        if ($image_type === \IMAGETYPE_PNG || $image_type === \IMAGETYPE_GIF) {
+            imagealphablending($tmp, false);
+            imagesavealpha($tmp, true);
+            $transparent = imagecolorallocatealpha($tmp, 255, 255, 255, 127);
+            imagefilledrectangle($tmp, 0, 0, $tn_width, $tn_height, $transparent);
+        }
+        imagecopyresampled($tmp, $src, 0, 0, 0, 0, $tn_width, $tn_height, $width, $height);
+
+        // create temporary thumb file
+        $thumbFile = tmpfile();
+        switch ($image_type) {
```

Comment:
I do not see any reason to use the same format as the source for thumbnail.

Reply:
addressed - now thumbnail format can be configured and it can be different than original image format